common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -365,7 +365,7 @@ protected boolean runAllTasks(long timeoutNanos) {
             // Check timeout every 64 tasks because nanoTime() is relatively expensive.
             // XXX: Hard-coded value - will make it configurable if it is really a problem.
             if ((runTasks & 0x3F) == 0) {
-                lastExecutionTime = ScheduledFutureTask.nanoTime();
+                lastExecutionTime = System.nanoTime();
                 if (lastExecutionTime >= deadline) {
```

Comment:
@normanmaurer - Yes

timeoutNanos <- 10
System.nanoTime() <- Long.MAX_VALUE - 5
deadline <- Long.MAX_VALUE + timeoutNanos (negative value)

using `if (System.nanoTime() >= deadline)` to detect if the current future has expired will be incorrect until `System.nanoTime()` rolls over and becomes negative.